reactivesocket-core/src/main/java/io/reactivesocket/internal/RemoteReceiver.java

```diff
@@ -176,6 +186,16 @@ public void cancel() {
         onError(new CancelException("Remote subscription cancelled."));
     }
 
+    private boolean isMissed() {
```

Comment:
Hmm, it's actually a bug introduced by my refactoring of `isMissed()` logic into a method.

There is a race between concurrent `subscribe()` and `on*()` setting `missed*` variables. These operations have to be atomic i.e. `subscription == null` and `missed*` variable initialization as `subscribe()` depends on it for properly invoking `subscriber`:

```java
synchronized(this) {
...
_missed = missedComplete || null != missedError;
...
}
if (_missed) {
...
``` 

Thanks!